inc/parser/handler.php

```diff
@@ -357,6 +357,100 @@ function quote($match, $state, $pos) {
         return true;
     }
 
+    /**
+     * Internal function for parsing highlight options.
+     * $options is parsed for key value pairs separated by commas.
+     * A value might also be missing in which case the value will simple
+     * be set to true. Commas in strings are ignored, e.g. option="4,56"
+     * will work as expected and will only create one entry.
+     *
+     * @param string $options Comma separated list of key-value pairs,
+     *                        e.g. option1=123, option2="456"
+     * @return array|null     Array of key-value pairs $array['key'] = 'value';
+     *                        or null if no entries found
+     */
+    protected function parse_highlight_options ($options) {
+        $result = array();
+        $concatenated = '';
+        preg_match_all('/(\w+(?:="[^"]*"))|(\w+[^=,\]])(?:,*)/', $options, $matches, PREG_SET_ORDER);
+        foreach ($matches as $match) {
+            $equal_sign = strpos($match [0], '=');
+            if ($equal_sign === false) {
+                $key = trim($match[0],',');
+                $result [$key] = true;
+            } else {
+                $key = substr($match[0], 0, $equal_sign);
+                $value = substr($match[0], $equal_sign+1);
+                $value = trim($value, '"');
+                $result [$key] = $value;
+            }
+        }
+
+        // Check for supported options
```

Comment:
I think the block below could be condensed even more.

First remove all unknown keys (see http://stackoverflow.com/a/4260168/172068):

```php
$result = array_intersect_key(
    $result,
    array_flip(
        'enable_line_numbers',
        'start_line_numbers_at',
        'highlight_lines_extra',
        'enable_keyword_links'
    )
);
```
Then sanitize the values:

```php
if(isset($result['enable_line_numbers'])) {
    $result['enable_line_numbers'] = (int) (bool) $result['enable_line_numbers'];
}
if(isset($result['highlight_lines_extra'])) {
    $result['highlight_lines_extra'] = array_map('intval', explode(',', $result['highlight_lines_extra']);
    $result['highlight_lines_extra'] = array_filter($result['highlight_lines_extra']);
    $result['highlight_lines_extra'] = array_unique($result['highlight_lines_extra']);
}
....
```
(code above untested and incomplete)

Not sure if it will actually end up much shorter, but it should be less complex than the foreach-switch-case-if-switch-case nestings used below.